### PR TITLE
Support to geojson compatible output #118

### PR DESCRIPTION
**Pull request by narigondelsiglo**
We have implemented geojson output support, but didn't write any test cases...
Don't know how to do it...
Hope it's helpful

### CODE REVIEW

File: lib/template/address-geojson.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    if (isset($aPlace['askml']))
+    {
+        $aFilteredPlaces['geokml'] = $aPlace['askml'];
+    }
+
+}
+
+javascript_renderData(array(
+    "type" => "Feature", "properties" => $aFilteredPlaces, "geometry" => array("type"=>"Point","coordinates"=>array($aFilteredPlaces['lon']*1,$aFilteredPlaces['lat']*1))
+));
```
Comment (lonvia) [inline lib/template/address-geojson.php]: You need to render a different array here when there is an error, lat and lon will not be set.

File: lib/template/search-geojson.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
+<?php
+
+$aFilteredPlaces = array();
+
+foreach($aSearchResults as $iResNum => $aPointDetails)
+{
+
+    $aPlace = array(
+            'place_id'=>$aPointDetails['place_id'],
+            'licence'=>"Data © OpenStreetMap contributors, ODbL 1.0. http://www.openstreetmap.org/copyright",
```
Comment (lonvia) [inline lib/template/search-geojson.php]: I believe that GeoJSON allows to put that on the FeatureCollection? If so, that would be better than repeating it for every feature.

File: website/reverse.php [hunk 3]
```diff
@@ -29,7 +29,8 @@
 $fThreshold = $oParams->getFloat('polygon_threshold', 0.0);
 
 // Format for output
-$sOutputFormat = $oParams->getSet('format', array('html', 'xml', 'json', 'jsonv2'), 'xml');
+$sOutputFormat = $oParams->getSet('format', array('html', 'xml', 'json', 'jsonv2', 'geojson'), 'xml');
+$bAsGeoJSON = $bAsGeoJSON || $sOutputFormat === 'geojson';
```
Comment (lonvia) [inline website/reverse.php]: This always downloads the polygon geometry. Can you make it like the other formats, where you just get the centroid point, if a polygon parameter is not explicitly given?

File: website/search.php [hunk 4]
```diff
@@ -26,15 +26,15 @@
...
+$sOutputFormat = $oParams->getSet('format', array('html', 'xml', 'json', 'jsonv2', 'geojson'), 'html');
 
 // Show / use polygons
 if ($sOutputFormat == 'html') {
     $oGeocode->setIncludePolygonAsGeoJSON($oParams->getBool('polygon_geojson'));
     $bAsGeoJSON = false;
 } else {
     $bAsPoints = $oParams->getBool('polygon');
-    $bAsGeoJSON = $oParams->getBool('polygon_geojson');
+    $bAsGeoJSON = $oParams->getBool('polygon_geojson') || $sOutputFormat === 'geojson';
```
Comment (lonvia) [inline website/search.php]: same here

File: lib/template/address-geojson.php (hunk 1 above)
Comment (narigondelsiglo) [inline lib/template/address-geojson.php]: Right!

File: lib/template/search-geojson.php (hunk 2 above)
Comment (narigondelsiglo) [inline lib/template/search-geojson.php]: Totally agree

File: website/reverse.php (hunk 3 above)
Comment (narigondelsiglo) [inline website/reverse.php]: Ok

File: website/search.php (hunk 4 above)
Comment (narigondelsiglo) [inline website/search.php]: Ok

**Comment by lonvia** [comment]
Closing in favour of #978.

**Pull request closed without merging**